exp/api/v1beta1/gcpmanagedcontrolplane_types.go

```diff
@@ -27,13 +27,241 @@ const (
+	// This setting is permanent.
+	// +optional
+	CidrBlock string `json:"cidrBlock,omitempty"`
+}
+
+// ClusterNetwork define the cluster network.
+type ClusterNetwork struct {
+	// PrivateCluster defines the private cluster spec.
+	// +optional
+	PrivateCluster PrivateCluster `json:"privateCluster,omitempty"`
+
+	// UseIPAliases is whether alias IPs will be used for pod IPs in the cluster. If false, routes will be used for
+	// pod IPs in the cluster.
+	// +optional
+	UseIPAliases bool `json:"useIPAliases,omitempty"`
+
+	// AuthorizedNetwork provide an IP-based firewall that controls access to the GKE control plane.
+	// +optional
+	AuthorizedNetwork AuthorizedNetwork `json:"authorizedNetwork,omitempty"`
+
+	// Pod defines the range of CIDRBlock list from where it gets the IP address.
+	// +optional
+	Pod ClusterNetworkPod `json:"pod,omitempty"`
+
+	// Service defines the range of CIDRBlock list from where it gets the IP address.
+	// +optional
+	Service ClusterNetworkService `json:"service,omitempty"`
+}
+
+// WorkloadIdentityConfig allows workloads in your GKE clusters to impersonate Identity and Access Management (IAM)
+// service accounts to access Google Cloud services.
+type WorkloadIdentityConfig struct {
+	// Enable is to enable the workload identity config.
+	// +optional
+	Enable bool `json:"enable,omitempty"`
+
+	// WorkloadPool is the workload pool to attach all Kubernetes service accounts to Google Cloud services.
+	// Only relevant when enabled is true
+	// +optional
+	WorkloadPool string `json:"workloadPool,omitempty"`
+}
+
+// AuthenticatorGroupConfig is RBAC security group for use with Google security groups in Kubernetes RBAC.
+type AuthenticatorGroupConfig struct {
+	// Enable is to enable the authenticator group config.
+	// +optional
+	Enable bool `json:"enable,omitempty"`
+
+	// SecurityGroups is the name of the security group-of-groups to be used.
+	// +optional
+	SecurityGroups string `json:"securityGroups,omitempty"`
+}
+
+// ClusterSecurity defines the cluster security.
+type ClusterSecurity struct {
+	// WorkloadIdentityConfig allows workloads in your GKE clusters to impersonate Identity and Access Management (IAM)
+	// service accounts to access Google Cloud services
+	// +optional
+	WorkloadIdentityConfig WorkloadIdentityConfig `json:"workloadIdentityConfig,omitempty"`
+
+	// AuthenticatorGroupConfig is RBAC security group for use with Google security groups in Kubernetes RBAC.
+	// +optional
+	AuthenticatorGroupConfig AuthenticatorGroupConfig `json:"authenticatorGroupConfig,omitempty"`
+
+	// EnableLegacyAuthorization Whether the legacy (ABAC) authorizer is enabled for this cluster.
+	// +optional
+	EnableLegacyAuthorization bool `json:"enableLegacyAuthorization,omitempty"`
+
+	// IssueClientCertificate is weather to issue a client certificate.
+	// +optional
+	IssueClientCertificate bool `json:"issueClientCertificate,omitempty"`
+}
+
+// AddonsConfig defines the enabled Cluster Addons.
+type AddonsConfig struct {
```

Comment:
Do we need to consider the other options:

https://cloud.google.com/go/docs/reference/cloud.google.com/go/container/latest/apiv1/containerpb#cloud_google_com_go_container_apiv1_containerpb_WorkloadIdentityConfig

Reply:
Sorry, I don't understand this :(